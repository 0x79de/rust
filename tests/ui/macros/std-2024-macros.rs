// Tests a small handful of macros in the standard library how they handle the
// new behavior introduced in 2024 that allows `const{}` expressions.

fn main() {
    assert_eq!(0, const { 0 });
    //~^ ERROR: no rules expected keyword `const`
    assert_eq!(const { 0 }, const { 0 });
    //~^ ERROR: no rules expected keyword `const`
    assert_eq!(const { 0 }, 0);
    //~^ ERROR: no rules expected keyword `const`

    let _: Vec<Vec<String>> = vec![const { vec![] }];
    //~^ ERROR: no rules expected keyword `const`
    let _: Vec<Vec<String>> = vec![const { vec![] }; 10];
    //~^ ERROR: no rules expected keyword `const`
}
