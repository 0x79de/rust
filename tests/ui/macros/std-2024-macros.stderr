error: no rules expected keyword `const`
  --> $DIR/std-2024-macros.rs:5:19
   |
LL |     assert_eq!(0, const { 0 });
   |                   ^^^^^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$right:expr`
  --> $SRC_DIR/core/src/macros/mod.rs:LL:COL

error: no rules expected keyword `const`
  --> $DIR/std-2024-macros.rs:7:16
   |
LL |     assert_eq!(const { 0 }, const { 0 });
   |                ^^^^^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$left:expr`
  --> $SRC_DIR/core/src/macros/mod.rs:LL:COL

error: no rules expected keyword `const`
  --> $DIR/std-2024-macros.rs:9:16
   |
LL |     assert_eq!(const { 0 }, 0);
   |                ^^^^^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$left:expr`
  --> $SRC_DIR/core/src/macros/mod.rs:LL:COL

error: no rules expected keyword `const`
  --> $DIR/std-2024-macros.rs:12:36
   |
LL |     let _: Vec<Vec<String>> = vec![const { vec![] }];
   |                                    ^^^^^ no rules expected this token in macro call
   |
   = note: while trying to match end of macro

error: no rules expected keyword `const`
  --> $DIR/std-2024-macros.rs:14:36
   |
LL |     let _: Vec<Vec<String>> = vec![const { vec![] }; 10];
   |                                    ^^^^^ no rules expected this token in macro call
   |
   = note: while trying to match end of macro

error: aborting due to 5 previous errors

